tests/integration/shared/sources/new-or-changed-sources-test.js

```diff
@@ -0,0 +1,195 @@
+const path = require('path')
+const fs = require('fs')
+const test = require('tape')
+
+const srcShared = path.join(process.cwd(), 'src', 'shared')
+const datetime = require(path.join(srcShared, 'datetime', 'index.js'))
+const sourceMap = require(path.join(srcShared, 'sources', '_lib', 'source-map.js'))
+const changedSources = require(path.join(__dirname, '_lib', 'changed-sources.js'))
+
+const srcEvents = path.join(process.cwd(), 'src', 'events')
+const crawlerHandler = require(path.join(srcEvents, 'crawler', 'index.js')).handler
+const scraperHandler = require(path.join(srcEvents, 'scraper', 'index.js')).handler
+
+/** Test new or changed scrapers.
+ *
+ * By default, this checks new/changed sources by diffing the current branch
+ * with a baseline branch (see _lib/changed-sources.js).
+ *
+ * You can override this by setting some environment vars:
+ *
+ * TEST_ALL - tests all of the sources
+ * e.g., TEST_ALL=1 npm run test:integration
+ *
+ * TEST_ONLY - a comma-delimited list of sources to test
+ * TEST_ONLY=gb-sct,nl,gb-eng npm run test:integration
+ */
+
+process.env.NODE_ENV = 'testing'
+
+// A fake cache, destroyed and re-created for the test run.
+process.env.LI_CACHE_PATH = path.join(process.cwd(), 'zz-testing-fake-cache')
+
+
+//////////////////////////////////////////////////////////////////////
+// Utilities
+
+/** Split an array into batches, e.g.:
+ * makeBatches([1,2,3,4,5], 3) = [[1,2,3], [4,5]]
+ */
+function makeBatches(arr, batchsize) {
+  const ret = []
+  for (let i = 0; i < arr.length; i += batchsize)
+    ret.push(arr.slice(i, i + batchsize))
+  return ret
+}
+
+/** Runs the full crawl-scrape-save cycle for a given source,
+ * returning a struct indicating which steps succeeded. */
+async function runCrawlAndScrape(key, today) {
+  const result = {
+    key: key,
```

Comment:
`key,` is sufficient fwiw!

Reply:
A lot of these comments are begging for lint `--fix`es ;)

Reply:
I like getting the detail in the response 😄 b/c it's fun. 
I thought I'd run `--fix` already ... will try again.  Thanks!

Reply:
`npm run lint --fix` didn't `fix` anything. :-P

Reply:
@camjc don't worry, we'll def add linting rules as necessary / appropriate! Just preferably not as blanket sets of hundreds at once.